### DISCUSSION TITLE
Question about SRFI-29 and changes to format

### DISCUSSION THREAD

**Discussion by jpellegrini** [egallesio/STklos#184] [General]
I have SRFI-29 implemented, except for one tiny bit: it requires a change to `format`.

What the SRFI wants is this: the code `~N@*` refers to the `N`-th argument after the format string. This is so messages in different languages can be formatted according to different orderings:
```
 (format "-- ~2@*~a ~a ~a --~%" 'x 'y 'z)
"-- z x y -- \n"
```
I see two ways of doing this:
* Changing the `internal_format` function in STklos
* Doing this in the SRFI:

```
(select-module STklos)

(import SRFI-29)

(define %stklos-format format)
(define format srfi-29:format)

(provide "srfi-29")
```

And `srfi-29:format` would translate the format string, removing all the `~N@*` and duplicating the format arguments where needed. Not trivial, since each time we deal with one of these new special format directive, we need to know all the other format directives that were behind it (so a simple regexp substitution won't work, it's somewhat trickier than that).

What do you think?

**Comment by jpellegrini** [comment]
Hi @egallesio . Never mind, I realized it wouldn't be good to change the C format code. I'll make the PR (but there is a glitch - see there)

**Comment by egallesio** [comment]
Hi @jpellegrini ,

Good idea to use Github discussions for ... "discussing".
